client/my-sites/site-settings/redirect-to-general.jsx

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import page from 'page';
+import { flowRight } from 'lodash';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isJetpackSite } from 'state/sites/selectors';
+import { isSiteAutomatedTransfer } from 'state/selectors';
+
+const redirectNonJetpackToGeneral = ( ComponentToRender ) => {
+	class RedirectNonJetpackToGeneral extends Component {
+		static propTypes = {
+			siteSlug: PropTypes.string,
+		}
+
+		componentDidMount() {
+			this.verifySiteIsJetpack();
+		}
+
+		componentDidUpdate() {
+			this.verifySiteIsJetpack();
+		}
+
+		verifySiteIsJetpack() {
+			if ( this.props.siteIsJetpack === false || this.props.siteIsAtomic ) {
+				this.redirectToGeneral();
+			}
+		}
+
+		redirectToGeneral = () => {
+			const {
+				siteId,
```

Comment:
Seems like you're using `siteId` here, but you've missed to add it to the connected props. You can do that, but you can also check for `siteSlug` which you're using below anyway.

Reply:
True! Great catch 👍